src/finch/tensor.py

```diff
@@ -380,19 +391,23 @@ def todense(self) -> np.ndarray:
             dense_tensor = dense_tensor.lvl
 
         result = np.asarray(jl.reshape(dense_tensor.val, shape))
-        return result.transpose(self.get_inv_order()) if self._is_dense else result
+        return (
+            result.transpose(tuple(o - 1 for o in self._order))
```

Comment:
This is the only concerning change to me. It can be very difficult to sufficiently test the difference between get_inv_order and o - 1 because we must use dimension != 2. I suppose your fix here is saying that when we densify, we aren't really changing the order at all so there's no reason to inv_order?

Reply:
With stacked matmul (implemented with lazy indexing) it seems that `get_inv_order` was wrongly used in the first place and `todense()` was broken. Previously we had only 2dim matmul by tensordot.

Reply:
See also #98 for a bug-fix and regression test.